core/src/utils/test/overlays/overlays.e2e.ts

```diff
@@ -254,5 +254,125 @@ configs({ modes: ['ios'], directions: ['ltr'] }).forEach(({ title, config }) =>
 
       await expect(modalInputOne).toBeFocused();
     });
+
+    test('focusing toast from a shadow overlay should return focus to the last focused element', async ({
+      page,
+      skip,
+    }) => {
+      test.info().annotations.push({
+        type: 'issue',
+        description: 'https://github.com/ionic-team/ionic-framework/issues/28261',
+      });
+      skip.browser('webkit', 'WebKit does not consider buttons to be focusable');
```

Comment:
Added in https://github.com/ionic-team/ionic-framework/pull/28950/commits/5d0121bd9eda801f9d618c30df58ba5ae2c5297c. I only did it for the shadow component tests (modal/popover). Doing the same for the scoped overlays is tricky because we don't have an easy way of connecting a non-button to an overlay since we are restricted in what we can display in these overlays.

However, I think the current coverage should be sufficient. In the event that this breaks due a browser bug, the shadow test should at least fail.